### ISSUE TITLE
Task Runner Explorer: Unexpected behavior with Gulp Watch task bound to Project Open

### ISSUE THREAD

**Issue by scottaddie** [aspnet/Tooling#373]
I created a new ASP.NET 5 (RC1 Update 1) StarterWeb project using VS 2015 Update 1. I then added the following watch task to `gulpfile.js`:

```javascript
gulp.task("watch", function () {
    gulp.watch([paths.css, "!" + paths.minCss], ["min:css"]);
    gulp.watch([paths.js, "!" + paths.minJs], ["min:js"]);
});
```

Next, I bound this "watch" task to Project Open and closed the solution. I opened the solution back up, and noticed 2 "watch (running)" tabs opened in TRX:

![watch_task](https://cloud.githubusercontent.com/assets/10702007/12375937/fa9188d0-bc9e-11e5-92af-83a8beae83be.png)

Why are there 2 tabs/running instances? I would've expected just 1.

Next, I unloaded the project in Solution Explorer via the "Unload Project" context menu item and the reloaded it via the "Reload Project" context menu item. The "watch" task ran again, and I now had 3 tabs/running instances:

![watch_task_after_reload](https://cloud.githubusercontent.com/assets/10702007/12375939/0934705a-bc9f-11e5-8ced-511b958f2c6e.png)

Why didn't the previous 2 tabs close upon unloading the project?

The "watch" task should be terminated when unloading the project.

**Comment by sayedihashimi** [comment]
TFS Bug#<a href="https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_workItems#id=231174&fullScreen=false&_a=edit">231174</a>

**Comment by jongunter** [comment]
I'm having this issue, too. I use a Git source control program outside of VS2015. Every time I switch to a different branch and the project reloads, a new "watch" task (which is bound to project open) starts up and the old one is not closed. 

If a dev switches branches several times during the day, it's very possible that 8 "watch" windows could be running if they forget to close them. That can cause performance issues on some machines

This is a pretty big issue, considering MSFT is pushing everyone towards a gulp/grunt based bundling/minification workflow now. I'm a little frustrated that this new workflow feels like it's so half-baked within the VS Tooling.

**Comment by MikhailTymchukDX** [comment]
@jongunter Yes, same for me. And sometimes previous watch task is killed and don't respawn at project open.
